src/signal_manager.sh

```diff
@@ -0,0 +1,66 @@
+include "$KW_LIB_DIR/kwio.sh"
+
+function get_valid_signals()
+{
+  local -r signal_list="$(trap -l | grep -oE 'SIG\S+')"
+  declare -gA SIGNAL_MANAGER_VALID_SIGNALS
+
+  for s in $signal_list; do
+    SIGNAL_MANAGER_VALID_SIGNALS["$s"]='1'
+  done
+}
+
+get_valid_signals
+
+# This functions prints the default message when interrupting kw
+function default_interrupt_handler()
+{
+  say $'\n'Oh no! An interruption! See ya...
+}
+
+# This function adds a new signal handler to an arbitrary signal
+#
+# @command The handler to be called when a signal is catched. This can
+#          can be a compound command (e.g. a function with parameters)
+#          but it must be passed as a single word (i.e. between quotes)
+# @signals The list of signals to attach to this handler
+#
+# If command is empty, use the default handler
+# If the signal is empty, use the default signals
+
+function signal_manager()
```

Comment:
I think the below shellcheck is a false positive. Just add `# shellcheck disable=SC2120` before this function declaration.